addons/account/tests/test_company_branch.py

```diff
@@ -235,3 +247,39 @@ def test_branch_should_keep_parent_company_currency(self):
         })
         self.env['account.chart.template'].try_loading('generic_coa', company=root_company.child_ids[0], install_demo=False)
         self.assertEqual(root_company.currency_id, root_company.child_ids[0].currency_id)
+
+    @users('branch_user')
+    def test_branch_user_can_assign_outstanding_credit_with_parent_access(self):
+        """Branch user (Billing) who can *see* both companies but keeps only
+        the branch active should still be able to reconcile an outstanding
+        credit without hitting AccessError."""
+
+        invoice = self.env['account.move'].with_context(allowed_company_ids=self.branch_a.ids).create({
+            'move_type':  'out_invoice',
+            'company_id': self.branch_a.id,
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2025-01-01',
+            'invoice_line_ids': [Command.create({
+                'name': 'Consulting service',
+                'price_unit': 100,
+                'tax_ids': False,
+            })],
+        })
+        invoice.action_post()
+
+        payment = self.env['account.payment'].with_context(allowed_company_ids=self.branch_a.ids).create({
+            'payment_type':   'inbound',
+            'partner_type':   'customer',
+            'partner_id':     self.partner_a.id,
+            'amount':         invoice.amount_total,
+            'journal_id':     self.company_data['default_journal_bank'].id,
+            'payment_method_line_id': self.inbound_payment_method_line.id,
+            'company_id':     self.branch_a.id,
+        })
+        payment.action_post()
+
+        self.root_company._invalidate_cache()
```

Comment:
why wouldn't call the private method

```suggestion
       self.root_company.invalidate_recordset(['tax_exigibility'], flush=False)
```